sklearn/decomposition/dict_learning.py

```diff
@@ -493,6 +494,12 @@ def dict_learning(X, n_components, alpha, max_iter=100, tol=1e-8,
 
         .. versionadded:: 0.20
 
+    method_max_iter : int, optional (default=1000)
+        It is passed to the underlying ``method`` as their ``max_iter``
```

Comment:
"Maximum number of iterations to perform it..."

Reply:
You mean to add this at the beginning of the sentence? I would become `Maximum number of iterations to perform it is passed to the underlying ``method`` as their ``max_iter`` parameter.`. I can't parse the sentence, confused!

Reply:
Since this is passed to `sparse_encode` I was thinking of using the docstring there here:

```py
method_max_iter : int, 1000 by default
    Maximum number of iterations to perform if `algorithm='lasso_cd'`.
```

This way a user would not need to figure it out by parsing the code to get to `sparse_code` and figure out how `max_iter` is used.

Reply:
I think this looks better now.